### DISCUSSION TITLE
GRPC node app backend Discussion!(Eucossa -2022)

### DISCUSSION THREAD

**Discussion by forinda** [EU-web-eucossa/inventory-service-nodejs-2022-s1#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by forinda** [comment]
I am working on setting up the Nodejs client for our `GRPC` and seemingly the client isn't picking the `stubs

@mwanagenzi @ElgarMokaya @Salomegit @Larrytaryan I need you to help me fix the issue below
Proto file definitions
```proto
syntax = "proto3";

message News {
    string id = 1;
    string title = 2;
    string body = 3;
    string postImage = 4;
}

service NewsService {
    rpc GetAllNews (Empty) returns (NewsList) {}
    rpc AddNews (News) returns (News) {}
}

message Empty {}

message NewsList {
   repeated News news = 1;
}
```
```sh
/home/orinda/Documents/dev/eucossa-products/src/grpc/client.js:22
        client.GetAllnews({}, (err, response) => {
               ^

TypeError: client.GetAllnews is not a function
    at Object.<anonymous> (/home/orinda/Documents/dev/eucossa-products/src/grpc/client.js:22:9)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
    at Module.load (node:internal/modules/cjs/loader:1033:32)
    at Function.Module._load (node:internal/modules/cjs/loader:868:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:47
[nodemon] app crashed - waiting for file changes before starting...
```

**Reply by mwanagenzi** [reply]
@forinda I'm not asking this as a cliche, but have you checked the docs for client definition on generating the stubs from proto files. There could be a missing keyword(s)

**Reply by forinda** [reply]
You can clone the repo and try solving in case I missed anything. 
I'm currently working on resolving the bugs

**Reply by mwanagenzi** [reply]
From the error, the function name in the proto file and the stub generated are different i.e "client.GetAllnews" is different from "GetAllNews" as you have defined in the proto file. Try checking that

**Reply by forinda** [reply]
@mwanagenzi Can you take this up as a bug issue?

**Reply by forinda** [reply]
Problem solved

**Reply by mwanagenzi** [reply]
Oh. Great! How did you solve it?

**Reply by forinda** [reply]
For the server had to make the proto service method names match the server method names
Proto
```proto
service NewsService {
    rpc GetAllNews (Empty) returns (NewsList) {}
    rpc AddNews (News) returns (News) {}
}
```
Server
```js
server.addService(newsProto.NewsService.service, {
	GetAllNews:async (call, callback) => {
		callback(null, { news });
	},
});
```

Client had issue with the naming
```diff
- client.GetAllnews({}, (err, response) => {
+ client.GetAllNews({}, (err, response) => {
		if (err) console.log(err);
		else console.log('Response: ', response);
});
```
So the issue was on the methosd being called `GetAllNews` as opposed to what I had which was `GetAllnews`

**Comment by mwanagenzi** [comment]
Hi. My name is Moses Ngure. I'm assisting in building this inventory service in Node.js
